data-access-layer/iaas/BackupStore.js

```diff
@@ -275,6 +276,22 @@ class BackupStore {
       .try(() => this.filename.isoDate(backupStartedBefore))
       .then(isoDate => this.listBackupFiles(options, getPredicate(isoDate), iteratees));
   }
+
+  listFilesOlderThan(options, dateOlderThan, container) {
+    const iteratees = ['lastModified'];
+    let prefix = options.prefix;
+
+    function getPredicate(isoDate) {
+      return function predicate(filenameobject) {
+        //transactionLogsDeletionStartDate defaults to current timestamp as part of isoDate function.
+        return _.lt(new Date(filenameobject.lastModified).toISOString(), isoDate);
+      };
+    }
+
+    return Promise
+      .try(() => this.filename.isoDate(dateOlderThan))
+      .then(isoDate => this.listFilenames(prefix, getPredicate(isoDate), iteratees, container, true));
```

Comment:
If you see, we have defined separate BackupStore for OOB and Service Instance where we do specific tasks like finding the prefix. I would recommend to have the same approach here (create a new BackupStore class for Service) and keep all iaas prefix related logic in iaas layer instead of caller. This whole function maybe then can be moved to separate BackupStore file, as I think it will be only used for transaction logs.

Reply:
The logic of introducing these functions is not limited to only Transaction Logs. I have explained this in a comment below.